### ISSUE TITLE
linter: optionally count TypeScript imports referenced by JSDoc `{@link}` / `{@linkcode}` as used in `no-unused-vars`

### ISSUE THREAD

**Issue by christopher-buss** [oxc-project/oxc#24996]
## Summary

Oxlint's native `eslint/no-unused-vars` reports a type-only import as unused when the imported symbol is referenced only by a JSDoc `{@link}` or `{@linkcode}` tag.

This is consistent with ESLint and typescript-eslint today, so this is an enhancement request rather than an ESLint-compatibility bug. An opt-in behavior would let projects preserve resolvable documentation links while using Oxlint's substantially faster native rule.

I searched the Oxc issue tracker for `no-unused-vars JSDoc`, `used only in JSDoc`, `@link unused`, and `JSDoc unused import`, but could not find an existing report.

## Environment

- Oxlint: 1.75.0
- Language: TypeScript
- Platform where reproduced: Windows

## Minimal reproduction

`box.ts`:

```ts
export interface Box {
	value: string;
}
```

`repro.ts`:

```ts
import type { Box } from "./box";

/**
 * Returns a {@link Box}.
 */
export function getBox(): unknown {
	return undefined;
}

/**
 * Creates a value compatible with {@linkcode Box}.
 */
export function createBox(): unknown {
	return undefined;
}
```

`.oxlintrc.json`:

```json
{
	"rules": {
		"no-unused-vars": "error"
	}
}
```

## Actual behavior

Oxlint reports `Box` as imported but never used:

```text
eslint(no-unused-vars): Type 'Box' is imported but never used.
```

Removing the import can break the intended documentation link. TypeDoc resolves `{@link}` through TypeScript resolution by default, using symbols in scope, and resolves `{@linkcode}` in the same way. Therefore, a type-only import can be meaningful to generated documentation even when it is absent from executable code and TypeScript type syntax. See the [TypeDoc `{@link}` documentation](https://typedoc.org/documents/Tags.__link_.html).

## Requested behavior

Please consider an opt-in option that counts imports referenced by resolvable JSDoc links as used. For example:

```json
{
	"rules": {
		"no-unused-vars": [
			"error",
			{
				"ignoreImportsUsedInJSDoc": true
			}
		]
	}
}
```

With that option enabled, `Box` would not produce a diagnostic in the reproduction above. With the option omitted or set to `false`, current behavior would remain unchanged.

The initial scope could be limited to imported symbols referenced by `{@link}`, `{@linkcode}`, or `{@linkplain}`. A broader option such as `countJSDocReferencesAsUsed` may be more appropriate if the implementation can reliably resolve other JSDoc symbol references.

## Comparison with other rules

`eslint-js/no-unused-vars` reports the same import. This matches ESLint core's documented definition of a variable use, which covers code-level reads, calls, construction, and argument passing but does not include JSDoc references. See the [ESLint `no-unused-vars` documentation](https://eslint.org/docs/latest/rules/no-unused-vars).

typescript-eslint explicitly documents that JSDoc `{@link}` references are unsupported by its `no-unused-vars` rule and shows an equivalent type-import example. See the [typescript-eslint `no-unused-vars` FAQ](https://typescript-eslint.io/rules/no-unused-vars/#why-are-variables-reported-as-unused-despite-being-referenced-by-link-in-jsdoc).

The eslint-plugin-unused-imports supports this via `isUsedInJsdoc`. See [here](https://github.com/sweepline/eslint-plugin-unused-imports/blob/853a372a6713c82283ed124c9eb073953072a76c/src/rules/predicates.ts#L15). This could be used as a basis.

## Motivation

The practical alternatives are currently:

1. Remove the import and lose or weaken the documentation link.
2. Add a suppression for every documentation-only import.
3. Run an ESLint-compatible JavaScript rule or plugin.

In a real project, the native rule completed in roughly 0.6 seconds while `eslint-js/no-unused-vars` took roughly 1.9 seconds over the same source tree. The JavaScript rule therefore adds significant lint overhead while still producing the same diagnostic. A native opt-in would retain Oxlint's performance advantage and avoid repeated suppressions for imports that serve a documentation purpose.

Oxlint documents `eslint/no-unused-vars` as applying to declarations, imports, and type declarations not used in code, and its current options do not include JSDoc handling. See the [Oxlint rule documentation](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-unused-vars.html) and [native rule source](https://github.com/oxc-project/oxc/blob/64c224117c2946ba4f1d1c7eb530bdf5bbbb61ae/crates/oxc_linter/src/rules/eslint/no_unused_vars.rs).

_Generated using Codex, reviewed and approved by a human (me!)_

**Comment by jaens** [comment]
#11639 - but the current issue has a more comprehensive description...

It's been requested quite a few times now.
